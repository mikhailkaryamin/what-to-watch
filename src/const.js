const AppRoute = {
  ADD_COMMENT: `/films/:id/review`,
  FILM: `/films/:id`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  PLAYER: `/player/:id`,
  ROOT: `/`,
};

const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ButtonType = {
  ADD_REVIEW: `ADD_REVIEW`,
  PLAY: `play`,
  LIST: `list`,
};

const EscKeyName = {
  LONG: `Escape`,
  SHORT: `Esc`,
};

const FilmCardsListType = {
  MAIN: `main`,
  LIKE_THIS: `like-this`,
};

const FilmDetailedTabsType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

const SpaceKeyName = {
  CODE: 32,
  TEXT: `Space`,
};

export {
  AppRoute,
  AuthStatus,
  ButtonType,
  EscKeyName,
  FilmCardsListType,
  FilmDetailedTabsType,
  SpaceKeyName,
};
