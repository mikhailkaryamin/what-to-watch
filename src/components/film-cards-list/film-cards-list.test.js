import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {FilmCardsListType} from '../../const.js';

import {
  initialState,
} from '../../mocks/initialState.js';

import FilmCardsList from './film-cards-list.jsx';

const mockStore = configureStore([]);

describe(`Film cards list`, () => {
  const store = mockStore(initialState);

  test(`should render for like this`, () => {
    const wrapper = renderer
    .create(
        <Provider store={store}>
          <FilmCardsList
            sign={FilmCardsListType.LIKE_THIS}
          />
        </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test(`should render for main`, () => {
    const wrapper = renderer
    .create(
        <Provider store={store}>
          <FilmCardsList
            sign={FilmCardsListType.MAIN}
          />
        </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
