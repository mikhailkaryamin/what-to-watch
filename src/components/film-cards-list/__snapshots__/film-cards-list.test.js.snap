// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film cards list should render for like this 1`] = `
<div
  className="catalog__movies-list catalog--like-this"
/>
`;

exports[`Film cards list should render for main 1`] = `
<div
  className="catalog__movies-list "
/>
`;
